packages/picker/src/Picker.ts

```diff
@@ -570,31 +595,39 @@ export class PickerBase extends SizedMixin(Focusable, { noDefaultSize: true }) {
                 // However, `appliesLabel` is applied by external elements that must be update complete as well to be bound appropriately.
                 await new Promise((res) => requestAnimationFrame(res));
                 await new Promise((res) => requestAnimationFrame(res));
-                if (
-                    !this.label &&
-                    !this.getAttribute('aria-label') &&
-                    !this.getAttribute('aria-labelledby') &&
-                    !this.appliedLabel
-                ) {
-                    window.__swc.warn(
-                        this,
-                        `<${this.localName}> needs one of the following to be accessible:`,
-                        'https://opensource.adobe.com/spectrum-web-components/components/picker/#accessibility',
-                        {
-                            type: 'accessibility',
-                            issues: [
-                                `an <sp-field-label> element with a \`for\` attribute referencing the \`id\` of the \`<${this.localName}>\`, or`,
-                                'value supplied to the "label" attribute, which will be displayed visually as placeholder text, or',
-                                'text content supplied in a <span> with slot="label", which will also be displayed visually as placeholder text.',
-                            ],
-                        }
-                    );
+                if (!this.hasAccessibleLabel) {
+                    this.warnNoLabel();
                 }
             });
         }
         super.update(changes);
     }
 
+    protected hasAccessibleLabel(): boolean {
```

Comment:
Action Menu was throwing the no label warning for a test that used the `label-only` slot, so I decided to override Picker's accessible label check and warning with a warning specific to action menu.